tuxemon/states/world/worldstate.py

```diff
@@ -1218,20 +1218,25 @@ def load_map(self, path: str) -> TuxemonMap:
         """
         txmn_map = TMXMapLoader().load(path)
         yaml_path = path[:-4] + ".yaml"
-        # TODO: merge the events from both sources
+        new_events = list(txmn_map.events)
+        new_inits = list(txmn_map.inits)
         if os.path.exists(yaml_path):
-            new_events = list(txmn_map.events)
-            new_events.extend(YAMLEventLoader().load_events(yaml_path))
-            txmn_map.events = new_events
+            new_events.extend(
+                YAMLEventLoader().load_events(yaml_path, "event")
+            )
+            new_inits.extend(YAMLEventLoader().load_events(yaml_path, "init"))
         # scenario YAML, try because not all maps have a scenario
-        try:
-            scenario_path = prepare.fetch("maps", txmn_map.scenario + ".yaml")
-            if os.path.exists(scenario_path):
-                new_events = list(txmn_map.events)
-                new_events.extend(YAMLEventLoader().load_events(scenario_path))
-                txmn_map.events = new_events
-        except:
-            pass
+        if txmn_map.scenario:
```

Comment:
this is probably fine for now, but it is a bit wastful as it forces the yaml loader to load and parse the file twice.  would probably be cleaner to return either a dictionary or a struct-type value with the events sorted into "events" and "inits" groups.

Reply:
you don't need to make this change now if you don't feel like it.  i don't expect it to be an issue since our map logic isn't very complex at the moment

Reply:
yeah, I hope this version is better